integration/tests/observer/execution_state_sync_test.go

```diff
@@ -0,0 +1,237 @@
+	an := s.net.ContainerByID(s.bridgeID)
+	eds, ctx := s.getTestExecutionDataServiceNode(an)
+
+	// setup storage objects needed to get the execution data id
+	db, err := an.DB()
+	require.NoError(s.T(), err, "could not open db")
+
+	metrics := metrics.NewNoopCollector()
+	headers := storage.NewHeaders(metrics, db)
+	results := storage.NewExecutionResults(metrics, db)
+
+	// Loop through checkBlocks and verify the execution data was downloaded correctly
+	for i := blockA.Header.Height; i <= blockA.Header.Height+checkBlocks; i++ {
+		header, err := headers.ByHeight(i)
+		require.NoError(s.T(), err, "could not get header")
+
+		result, err := results.ByBlockID(header.ID())
+		require.NoError(s.T(), err, "could not get sealed result")
+
+		s.T().Logf("getting execution data for height %d, block %s, execution_data %s", header.Height, header.ID(), result.ExecutionDataID)
+
+		ed, err := eds.Get(ctx, result.ExecutionDataID)
+		assert.NoError(s.T(), err, "could not get execution data for height %v", i)
+
+		s.T().Logf("got execution data for height %d", i)
+		assert.Equal(s.T(), header.ID(), ed.BlockID)
+	}
+}
+
+func (s *ExecutionStateSyncSuite) getTestExecutionDataServiceNode(node *testnet.Container) (state_synchronization.ExecutionDataService, irrecoverable.SignalerContext) {
+	ctx, errChan := irrecoverable.WithSignaler(s.ctx)
+	go func() {
+		select {
+		case <-s.ctx.Done():
+			return
+		case err := <-errChan:
+			s.T().Errorf("irrecoverable error: %v", err)
+		}
+	}()
+
+	ds, err := badgerds.NewDatastore(node.ExecutionDataDBPath(), &badgerds.DefaultOptions)
+	require.NoError(s.T(), err, "could not get execution datastore")
+
+	go func() {
+		<-s.ctx.Done()
+		if err := ds.Close(); err != nil {
+			s.T().Logf("could not close execution data datastore: %v", err)
+		}
+	}()
+
+	blobService := utils.NewLocalBlobService(ds, utils.WithHashOnRead(true))
+	blobService.Start(ctx)
+
+	return state_synchronization.NewExecutionDataService(
+		cbor.NewCodec(),
+		compressor.NewLz4Compressor(),
+		blobService,
+		metrics.NewNoopCollector(),
+		zerolog.New(os.Stdout).With().Str("test", "execution-state-sync").Logger(),
+	), ctx
+}
+
+func (s *ExecutionStateSyncSuite) startObserver(ctx context.Context) {
+	stop, err := s.net.AddObserver(ctx, &testnet.ObserverConfig{
+		ObserverName:            "observer_1",
+		ObserverImage:           "gcr.io/flow-container-registry/observer:latest",
+		AccessName:              "access_1",
+		AccessPublicNetworkPort: fmt.Sprint(testnet.AccessNodePublicNetworkPort),
+		AccessGRPCSecurePort:    fmt.Sprint(testnet.DefaultSecureGRPCPort),
+	})
+
+	if err != nil {
+		// this can happen occaisionally...
+		// usually it's because docker didn't remove the observer container during the previous test.
+		// the observer container is removed by an "AutoRemove" flag in AddObserver.
+		panic(err)
+	}
+
+	time.Sleep(time.Second * 3) // needs breathing room for the observer to start listening
```

Comment:
you can add a simple function here to spin on port being open.  this will make this test slightly more robust w/o slowing it down.

(this function could likely be reused by many other integration tests.)